### ISSUE TITLE
Docker-S3 Push folder wrong value

### ISSUE THREAD

**Issue by santiago1989** [PrefectHQ/prefect#9458]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

At recipe `s3` / `docker-s3`, the push folder is not the same as for pull when `folder` field is provided.

### Reproduction

```python3
Executing the command `prefect project init --recipe docker-s3 -f bucket=s3-bucket -f folder=s3-folder` (also for `--recipe s3`), the result is:

name: git-repo
prefect-version: 2.10.7

# build section allows you to manage and build docker images
build:
- prefect_docker.projects.steps.build_docker_image:
    requires: prefect-docker>=0.2.0
    image_name: "{{ image_name }}"
    tag: "{{ tag }}"
    dockerfile: Dockerfile
    push: true

# push section allows you to manage if and how this project is uploaded to remote locations
push:
- prefect_aws.projects.steps.push_project_to_s3:
    requires: prefect-aws>=0.3.0
    bucket: s3-bucket
    folder: git-repo

# pull section allows you to provide instructions for cloning this project in remote locations
pull:
- prefect_aws.projects.steps.pull_project_from_s3:
    requires: prefect-aws>=0.3.0
    bucket: s3-bucket
    folder: s3-folder
```

### Versions

```Text
prefect: 2.10.7
```

**Comment by desertaxle** [comment]
The push/pull folder mismatch has been fixed in the current `docker-s3` recipe. The push step now uses `folder: "{{ name }}"` and the pull step references the push output with `folder: "{{ push_code.folder }}"`, so they're guaranteed to stay in sync.

Note that `folder` isn't a supported recipe input — the recipe always uses the project directory name. If you need a custom folder, you can edit the generated `prefect.yaml` directly after `prefect init`. If you're still seeing push/pull mismatches on a current version, please open a new issue.
